server/models/GeminiModel.ts

```diff
@@ -0,0 +1,111 @@
+import { getEnv } from '@/server/utils/environment';
+import { GoogleGenerativeAI } from '@google/generative-ai';
+
+import type { GenerativeModel, InputContent } from '@google/generative-ai';
+import type { AiInterface } from '@/server/types/index';
+
+/**
+ * Class representing the Gemini API connection and its methods.
+ * This is an implementation of the AiInterface.
+ */
+class Gemini implements AiInterface {
+  apiKey: string = this.setApiKey();
+  ai: GoogleGenerativeAI = this.connect();
+  model: GenerativeModel | undefined;
+
+  /**
+   * Create a Gemini API connection.
+   */
+  constructor() {
+    getEnv();
+  }
+
+  /**
+   * Set the Gemini API key using the environment variable.
+   *
+   * @return the Gemini API key
+   */
+  setApiKey(): string {
+    try {
```

Comment:
Since accessing an unset environment variable with process.env.VARIABLE_NAME returns undefined instead of throwing an error, the catch block won't be executed. If you want to ensure the presence of the GEMINI_API_KEY, use a fallback and an `if` state to directly check for empty string and throw an error.

Reply:
awesome catch, thank you.